File: PlusMinusResolved.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{
    public static int posi, neg, zero = 0;

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<int> arr)
    {
        // I check that it is not sending an empty list 
        if (arr.Count() <= 0 ) {
			Console.Write("You have not sent any data please check it");
			return;
		};
		
        //Values are assigned to the variables, which represent the different values
		for (int i = 0; i < arr.Count(); i++){
		  if(arr[i] < 0){
			  neg++;
		  } else if (arr[i] > 0){
			  posi++;
		  } else {
			  zero++;
		  }
		}
	  	
        //Formatted to floating point
		double _posi = (double) posi/arr.Count();
		double _neg = (double) neg/arr.Count();
		double _zero = (double) zero/arr.Count();
		
        //Converts to 6-place decimals using ToString and print
        Console.Write(_posi.ToString("0.000000")+"\n");
		Console.Write(_neg.ToString("0.000000")+"\n");
		Console.Write(_zero.ToString("0.000000")+"\n");
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> arr = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arrTemp => Convert.ToInt32(arrTemp)).ToList();

        Result.plusMinus(arr);
    }
}

File: PlusMinus_Resolved.php
<?php

/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus($arr) {
    // Write your code here
   
//we check that the argument is an array
   if (is_array($arr)) {
   	     //We count the array indices
		 $CountTot =  count($arr);
   } else {
   	  echo "Error - Argument is not a valid array";
   	  return ;
   } 

   //Initialize values
   $posi = $neg = $zero = 0;

   
   //Values are assigned to the variables, which represent the different values
   for($i = 0; $i < $CountTot; $i++){
       if($arr[$i] > 0){
           $posi++;
       }elseif($arr[$i] < 0){
           $neg++;
       }else{
           $zero++;
       }
   }

    //Formatted to floating point
   $_posi = (float) $posi/$CountTot;
   $_neg = (float) $neg/$CountTot;
   $_zero =  (float) $zero/$CountTot;
                             
    //Converts to 6-place decimals using PHP's number_format function
   $posi = number_format($_posi, 6, '.', '');
   $neg = number_format($_neg, 6, '.', '');
   $zero = number_format($_zero, 6, '.', '');

   //each variable is printed on its own line, according to instructions
   print($posi . "\n" . $neg . "\n" . $zero);  
    
}

$n = intval(trim(fgets(STDIN)));

$arr_temp = rtrim(fgets(STDIN));

$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

plusMinus($arr);
